src/orbit.rs

```diff
@@ -31,75 +36,67 @@ pub struct OrbitMetadata {
     pub write_delegators: Vec<DIDURL>,
     // TODO placeholder type
     pub revocations: Vec<String>,
+    #[serde(default)]
+    pub auth: AuthTypes,
 }
 
-mod cid_serde {
-    use libipld::cid::{multibase::Base, Cid};
-    use serde::{de::Error as SError, ser::Error as DError, Deserialize, Deserializer, Serializer};
-
-    pub fn serialize<S>(cid: &Cid, ser: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        ser.serialize_str(
-            &cid.to_string_of_base(Base::Base58Btc)
-                .map_err(S::Error::custom)?,
-        )
-    }
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(untagged, rename_all = "UPPERCASE")]
+pub enum AuthTypes {
+    Tezos,
+    ZCAP,
+}
 
-    pub fn deserialize<'de, D>(deser: D) -> Result<Cid, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        let s: &str = Deserialize::deserialize(deser)?;
-        s.parse().map_err(D::Error::custom)
+impl Default for AuthTypes {
+    fn default() -> Self {
+        Self::Tezos
     }
 }
 
-// TODO I think this will need to go, using a trait for a core object creates
-// too much monomorphisation which shouldn't be too much of a problem for the
-// binary size but I'm not sure how Rocket handles it.
-// Or maybe implement the traits separately.
-// Ultimately I think we'll need to use an enum, it's not as fancy but because
-// we only have simple relationships (e.g. for the Tezos policy you need a
-// Tezos token) it will make the routing simpler and easier to use.
-#[rocket::async_trait]
-pub trait Orbit: ContentAddressedStorage {
-    // + AuthorizationPolicy {
-    fn id(&self) -> &Cid;
-
-    fn hosts(&self) -> Vec<PeerId>;
-
-    fn admins(&self) -> &[&DIDURL];
-
-    fn auth(&self) -> &AuthMethods;
-
-    fn make_uri(&self, cid: &Cid) -> Result<String>;
-
-    // async fn update(
-    //     &self,
-    //     update: Self::UpdateMessage,
-    // ) -> Result<(), <Self as ContentAddressedStorage>::Error>;
+impl From<&AuthMethods> for AuthTypes {
+    fn from(m: &AuthMethods) -> Self {
+        match m {
+            AuthMethods::Tezos(_) => Self::Tezos,
+            AuthMethods::ZCAP(_) => Self::ZCAP,
+        }
+    }
 }
 
 #[derive(Clone)]
 pub enum AuthMethods {
     Tezos(TezosBasicAuthorization),
+    ZCAP(ZCAPAuthorization),
 }
 
 #[derive(Clone)]
 pub enum AuthTokens {
     Tezos(TezosAuthorizationString),
+    ZCAP(ZCAPTokens),
 }
 
-impl AuthorizationToken for AuthTokens {
-    fn extract(auth_data: &str) -> Result<Self> {
-        Err(anyhow!("todo"))
+#[rocket::async_trait]
+impl<'r> FromRequest<'r> for AuthTokens {
```

Comment:
Maybe in the idea of decoupling Rocket from the library the routing based on the headers should be centralised somewhere else (`auth.rs`?)